Add upstream collector howto (#433)
elastic/opentelemetry: docs/reference/edot-collector/upstream-collector.md

---
navigation_title: Use the upstream Collector
description: Learn how to send data to Elastic Observability using the upstream OpenTelemetry Collector instead of EDOT.
applies_to:
  stack:
  serverless:
    observability:
  product:
    edot_collector: ga
products:
  - id: cloud-serverless
  - id: observability
  - id: edot-collector
---

# Send data to {{serverless-full}} using the upstream Collector

> While the [{{edot}} (EDOT) Collector](/reference/index.md) provides a streamlined experience with pre-selected components, you can also use the contrib OpenTelemetry Collector or a custom distribution to send data to Elastic Observability. This approach requires more configuration but gives you equal control over your OpenTelemetry setup.

Author: True!

## Overview

The upstream OpenTelemetry Collector is the community-maintained version that provides the foundation for all OpenTelemetry distributions. To configure it to work with Elastic Observability, you need to:

- Manually select and configure components.
- Set up proper data processing pipelines.
- Handle authentication and connection details.
- Ensure compatibility with Elastic's data requirements.

## Deployment scenarios

> The configuration requirements vary depending on your Elastic deployment model. The following sections outline what you need for each scenario.

Reviewer: ```suggestion
The configuration requirements vary depending on your use case and the Elastic deployment model you want to send data to. The following sections outline what you need for each scenario.
```

### Elastic Cloud Serverless

> {{serverless-full}} provides a [Managed OTLP Endpoint](/reference/motlp.md) that accepts OpenTelemetry data in its native format. This makes it the simplest scenario for using upstream components because no special processing is needed.

Reviewer: ```suggestion
{{serverless-full}} provides a [Managed OTLP Endpoint](/reference/motlp.md) that accepts OpenTelemetry data in its native format. This makes it the simplest scenario for using upstream components because scaling and signal processing (eg. producing metrics from events) is handled by Elastic.
```

The following configuration example shows how to send data to the Managed OTLP Endpoint:

```yaml
exporters:
  otlpexporter:
    endpoint: "https://your-deployment.elastic-cloud.com:443"
    headers:
      authorization: "Bearer YOUR_API_KEY"

service:
  pipelines:
    traces:
      exporters: [otlpexporter]
    metrics:

>       exporters: [otlpexporter]

Reviewer: ```suggestion
      exporters: [otlp]
```
